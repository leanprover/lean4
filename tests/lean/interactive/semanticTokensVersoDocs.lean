set_option doc.verso true
/-!
This test checks that Verso docstring semantic tokens work as expected. In particular, it tests that
overlapping token handling does what we want, because the unannotated identifiers and the spaces in
the {lit}`code` elements are assigned the string type, while variables etc are given info-based
tokens.
-/
/-- {name}`foo1` {lean}`foo1 x` {assert}`foo1 4 = 5` -/
def foo1 (x : Nat) := x.succ
/-- {name}`foo1` {lean}`foo1 x` {assert}`foo2 = foo1` -/
def foo2 (x : Nat) := x |>.succ
/--
*bold* _emph_ *_both_* {lit}`code`
```leanTerm
(fun _ => rfl : âˆ€ y : Unit, x = y)
```
* List
* More list
  1. Nested list
  2. List

  : Term (nested)

    Description

# Header 1

## Header 2

[![link][url]][url]

[url]: http://example.com/example.gif

-/
def x := ()

--^ collectDiagnostics
--^ textDocument/semanticTokens/full
