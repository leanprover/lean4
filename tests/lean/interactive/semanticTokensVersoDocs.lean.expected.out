{"version": 1, "uri": "file:///semanticTokensVersoDocs.lean", "diagnostics": []}
{"textDocument": {"uri": "file:///semanticTokensVersoDocs.lean"},
 "position": {"line": 35, "character": 2}}
{"data":
 [0,
  0,
  10,
  0,
  0,
  0,
  21,
  4,
  0,
  0,
  4,
  4,
  1,
  0,
  0,
  0,
  1,
  3,
  3,
  0,
  0,
  3,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  4,
  18,
  0,
  0,
  4,
  1,
  0,
  0,
  3,
  4,
  1,
  0,
  0,
  0,
  1,
  4,
  3,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  4,
  18,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  4,
  3,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  5,
  18,
  0,
  0,
  5,
  1,
  1,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  6,
  3,
  0,
  0,
  6,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  10,
  18,
  0,
  0,
  10,
  1,
  0,
  0,
  1,
  0,
  3,
  0,
  0,
  0,
  10,
  1,
  1,
  0,
  0,
  12,
  1,
  1,
  0,
  0,
  2,
  4,
  2,
  0,
  1,
  4,
  1,
  0,
  0,
  0,
  1,
  4,
  3,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  4,
  18,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  4,
  3,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  5,
  18,
  0,
  0,
  5,
  1,
  1,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  6,
  3,
  0,
  0,
  6,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  11,
  18,
  0,
  0,
  11,
  1,
  0,
  0,
  1,
  0,
  3,
  0,
  0,
  0,
  10,
  1,
  1,
  0,
  0,
  12,
  1,
  1,
  0,
  0,
  5,
  4,
  2,
  0,
  2,
  0,
  1,
  0,
  0,
  0,
  5,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  5,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  5,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  2,
  1,
  0,
  0,
  0,
  1,
  3,
  3,
  0,
  0,
  3,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  4,
  18,
  0,
  0,
  4,
  1,
  0,
  0,
  1,
  0,
  3,
  0,
  0,
  0,
  3,
  8,
  3,
  0,
  1,
  0,
  18,
  18,
  0,
  0,
  18,
  1,
  1,
  0,
  0,
  1,
  13,
  18,
  0,
  0,
  13,
  1,
  1,
  0,
  0,
  1,
  1,
  18,
  0,
  1,
  0,
  0,
  18,
  0,
  0,
  0,
  3,
  0,
  0,
  1,
  0,
  1,
  0,
  0,
  1,
  0,
  1,
  0,
  0,
  1,
  2,
  2,
  0,
  0,
  1,
  2,
  2,
  0,
  0,
  2,
  2,
  1,
  0,
  0,
  4,
  0,
  1,
  0,
  0,
  2,
  0,
  2,
  0,
  0,
  2,
  0,
  1,
  0,
  0,
  0,
  1,
  2,
  0,
  0,
  0,
  2,
  4,
  18,
  0,
  0,
  4,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  3,
  2,
  0,
  0,
  3,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  3,
  2,
  0,
  0,
  3,
  1,
  0,
  0,
  2,
  0,
  1,
  0,
  0,
  0,
  1,
  3,
  2,
  0,
  0,
  3,
  2,
  0,
  0,
  0,
  3,
  30,
  18,
  0,
  3,
  0,
  3,
  0,
  0]}
